kinto_http/cli_utils.py

```diff
@@ -19,8 +19,12 @@ def get_auth(auth):
 
 def create_client_from_args(args):
     """Return a client from parser args."""
+    if args.bearer_token:
```

Comment:
What do you think of using `--auth abcedf` and `if ":" not in auth: auth = BearerToken(auth)` ?

Reply:
This change the current behaviour that if auth doesn't contain a `:` we ask for the password.

Also `BearerTokenAuth(auth)` is not sufficient most of the time because you might want to give a bearer-realm (in our case `Bearer+Email` or `Nexmo`)

The ultimate argument being from the Zen of Python: Explicit is better than implicit.

Reply:
I also like not to bloat CLI args. We have a ``--auth`` which is fine for passing the string.
What if we only add  ``--bearer-type=Nexmo``?

It becomes `--auth abcdef --bearer-type=Bearer` with `if args.bearer_type: auth = BearerToken(auth)`

Reply:
I was thinking about `--auth Bearer=TOKEN` at some point too. But it is really likely to be wrong. If you look at a = in the string it can as well be a character of the password or of the token.

I like the fact that there is not too much magic. --auth is for basicauth everything else is a different story.

We don't risk to break the current API plus we have a handfull of parameters so we are far from bloating it. Look at curl args if you are worried.

Reply:
What's wrong `--auth abcdef --bearer-type=Bearer`?

Reply:
What's wrong with `--bearer-token`?

Reply:
I tried and it is tricky to handle because of the AuthAction mecanism that is not supposed to rely on another args to take action.

Reply:
> What's wrong with --bearer-token?

It's 2 more CLI args

Reply:
I don't see a problem with the number of args, I am a bit lost in capturing what bothers you to be honest.

Reply:
Less is more etc. Less CLI args, less documentation, less to understand etc.

I am also a bit lost to grasp what's wrong with something like: `kinto-wizard --auth prout --auth-type=Bearer` or similar. 

Hey, *it's not a hill I want to die on* :)

Reply:
Personally, I also prefer having fewer command line arguments. (Less to support. Fewer things to read when you do `--help`.) We're defining our external API here so I feel it's worth taking a little time to get it right.

I feel that having an `auth` argument for one type of auth and other arguments for different types of auth is confusing. If we ask ourselves "who is the audience for this tool?" I think the answer is "web developers". This is especially true because Kinto tends to lean into the "just an HTTP API" paradigm. I think re-using the HTTP conventions is therefore the most natural thing. In this case, that means the `Authorization` header. So I'd propose just using `--auth`: `--auth username:password`, `--auth "Bearer abcdef"`, `--auth "Nexmo abcdef"`.

Reply:
Ok this last proposal makes a lot of sense to me thanks.

Reply:
Thanks Ethan for mediating this!
I like your proposal too :)